BUILDING.md

```diff
@@ -12,10 +12,10 @@ Before starting, note that ESBMC is mainly distributed under the terms of the [A
 
 | package   | required | minimum version | recommended version |
 |-----------|----------|-----------------|---------------------|
-| clang     | yes      | 11.0.0          |                     |
-| boost     | yes      | 1.77            |                     |
-| Boolector | no       | 2.3.2           |                     |
-| CVC4      | no       | 1.8             |                     |
+| clang     | yes      | 11.0.0          | latest              |
```

Comment:
I would put 11 as the recommended here. Its been tested for years now.

Reply:
OK, unsure what to. @mikhailramalho argued for users to switch to the latest version 14.0.x in #790, though maybe based on a misunderstanding of the API change. I've been testing clang-13 for half a year. Please advise.

Reply:
>  I've been testing clang-13 for half a year

Unix systems should be fine, the main problem is Windows. As of today, ESBMC will not compile with the default instalation of LLVM 14.

And by tested, I mean by external users + sv-comp.

> Please advise.

Since it is a requirement, we should not recommend LLVM 14 before we confirm that we can build every version of ESBMC with it.

@lucasccordeiro and @mikhailramalho what do you think?

Reply:
We have Intel as our main customer for the ESBMC Windows version. So I would support what Rafael suggests about the LLVM version.